### PR TITLE
Server: upvote downvote api

### PR DESCRIPTION
**Pull request by EugeneTeu**

### CODE REVIEW

File: server/src/routes/questions.ts [hunk 1]
```diff
@@ -64,30 +65,47 @@ router.post("/", verifyUserAuth, async (req: Request, res: Response) => {
...
+      userId,
+      questionId,
+      command,
+      VoteType.UPVOTE
+    );
+    const updatedQuestion: Question = await editUpvoteDownvoteQuestion(
+      userId,
+      questionId,
+      incObject
+    );
```
Comment (EugeneTeu) [inline server/src/routes/questions.ts:86]: this method doesnt care about anyhting, just based on incObject

File: server/src/controllers/votes.ts [hunk 2]
```diff
@@ -1,52 +1,97 @@
...
+    } else {
+      incObject.downvotes = -1;
+    }
+    return incObject;
   }
+  // if vote already present and type is correct
+  // if vote not present and type is remove
+  // we dont increment or decrement
+  return incObject;
+}
```
Comment (EugeneTeu) [inline server/src/controllers/votes.ts:82]: feel free to improve the logic here

Comment (EugeneTeu) [reply server/src/controllers/votes.ts:82]: quite complex cause i combined both upvote and downvotes

**Review by AaronCQL — APPROVED** [review]
LGTM

**Pull request merged**